pkg/lib/usage/globaldb_store.go

```diff
@@ -205,3 +205,50 @@ func (s *GlobalDBStore) FetchUsageRecords(
 
 	return out, nil
 }
+
+func (s *GlobalDBStore) FetchUsageRecordsInRange(
+	ctx context.Context,
+	appID string,
+	recordName RecordName,
+	period periodical.Type,
+	fromStartTime time.Time,
+	toEndTime time.Time,
+) ([]*UsageRecord, error) {
+	q := s.SQLBuilder.Select(
+		"id",
+		"app_id",
+		"name",
+		"period",
+		"start_time",
+		"end_time",
+		"count",
+		"stripe_timestamp",
+	).
+		From(s.SQLBuilder.TableName("_portal_usage_record")).
+		Where(
+			"app_id = ? AND name = ? AND period = ? AND start_time >= ? AND end_time <= ?",
```

Comment:
The index we have only cover `(app_id, name, period, start_time)`. If we use `end_time` in the query, the index may not be used.

However, `end_time` is actually determined by `period` and `start_time`. So I think we can rewrite the query so that it only reference the indexed columns.

Reply:
@louischan-oursky I've confirmed the index can be correctly used:
```
EXPLAIN ANALYZE SELECT * FROM _portal_usage_record WHERE app_id = 'my-app' AND name = 'active-user' AND period = '1 day' AND start_time >= NOW() AND end_time <= NOW();
```

| QUERY PLAN |
| --- |
| Index Scan using \_portal\_usage\_record\_app\_id\_name\_period\_start\_time\_key on \_portal\_usage\_record  \(cost=0.15..8.18 rows=1 width=188\) \(actual time=0.136..0.137 rows=0 loops=1\) |
|   Index Cond: \(\(app\_id = 'my\-app'::text\) AND \(name = 'active\-user'::text\) AND \(period = '1 day'::text\) AND \(start\_time \>= now\(\)\)\) |
|   Filter: \(end\_time \<= now\(\)\) |
| Planning Time: 0.732 ms |
| Execution Time: 0.244 ms |

Reply:
The `start_time >= t` condition can still include rows that is to be filtered by the filter. But we actually can rewrite the condition to `start_time >= t0 AND start_time < t1`

Reply:
Do you mean you want to skip the step `Filter: (end_time <= now())`?

Reply:
Sorry wait, it seems I've calculated the end time wrongly for monthly period.

Reply:
Fixed in the latest commit.